messaging-client-databus/src/main/java/com/inmobi/databus/partition/CollectorReader.java

```diff
@@ -295,12 +281,30 @@ public synchronized Long getReaderBackLog() throws IOException {
       }
       retryCount++;
     }
-    if (lReader.getCurrentFile() != null) {
-      localStreamPendingSize += lReader.getPendingSize();
-      LOG.info("Pending Size of compressed data inside streams_local output location starting from file: " +
-              lReader.getCurrentFile() + " is " + localStreamPendingSize);
+    /**
+     * Reader can be either local stream reader or collector reader at any point of time.
+     * Based on that, the other reader's pending size to be calculated, along with current reader's
+     * pending size. Return sum of both, after adjusting local stream reader's pending size
+     * with compression factor.
+     */
+    if (reader == lReader) {
+      String collectorPath = CollectorStreamReader.getCollectorFileName(streamName, reader.getCurrentFile().getName());
```

Comment:
- Split each code block into 2 parts: finding creader pending size and lreader pending size.
- Use specific reader to get current file and check for non null.